src/sankey.js

```diff
@@ -0,0 +1,152 @@
+import React, {PropTypes, Component, cloneElement} from 'react';
+import shallowCompare from 'react-addons-shallow-compare';
+import linkKey from './utils/link-key';
+import getWidthSums from './utils/get-width-sums';
+import separateChildrenByType from './utils/separate-children-by-type';
+import findColumns from './utils/find-columns';
+import configureNodeCoordinates from './utils/configure-node-coordinates';
+import getAnchorFromRectangleNodes from './utils/get-anchor-from-rectangle-nodes';
+
+const {number} = PropTypes;
+export default class Sankey extends Component {
+
+  static propTypes = {
+    /** width of svg figure */
+    width: number.isRequired,
+    /** height of svg figure */
+    height: number.isRequired,
+    /** spacing (horizontal) */
+    spacing: number,
+    /** margin (vertical) */
+    margin: number
+  };
+
+  shouldComponentUpdate(nextProp, nextState) {
+    return shallowCompare(this, nextProp, nextState);
+  }
+
+  render() {
+    const {
+      children,
+      width: containerWidth,
+      height: containerHeight,
+      spacing,
+      margin,
+      ..._props
+    } = this.props;
+
+    const {defs, nodes, links} = separateChildrenByType(children);
+
+    const columns = findColumns(nodes, links);
+
+    const defaultColumnWidth = (containerWidth - columns.length * spacing) /
+      (columns.length + 1);
+
+
+    /* if the max width of the column is less than the default column with,
+     * take the bigger value. When no width is given, a default column width
+     * is used. */
+    const columnWidths = columns.map(
+      column =>
+        (Math.max.apply(null, column.map(
+          ({props: {width, r}}) => width || r * 2)
+        ) || defaultColumnWidth)
```

Comment:
it can not be solved by default values, because the default column width needs the constructed column stack for it's calculation.